libs/execution_context/ebpf_maps.c

```diff
@@ -19,77 +21,33 @@ typedef struct _ebpf_map_function_table
     void (*delete_map)(_In_ ebpf_core_map_t* map);
     uint8_t* (*find_entry)(_In_ ebpf_core_map_t* map, _In_ const uint8_t* key);
     ebpf_result_t (*update_entry)(_In_ ebpf_core_map_t* map, _In_ const uint8_t* key, _In_ const uint8_t* value);
+    ebpf_result_t (*update_entry_with_handle)(
+        _In_ ebpf_core_map_t* map, _In_ const uint8_t* key, _In_ const uint8_t* value, uintptr_t value_handle);
     ebpf_result_t (*delete_entry)(_In_ ebpf_core_map_t* map, _In_ const uint8_t* key);
     ebpf_result_t (*next_key)(_In_ ebpf_core_map_t* map, _In_ const uint8_t* previous_key, _Out_ uint8_t* next_key);
 } ebpf_map_function_table_t;
 
-extern ebpf_map_function_table_t ebpf_map_function_tables[BPF_MAP_TYPE_ARRAY + 1];
-
-ebpf_result_t
-ebpf_map_create(_In_ const ebpf_map_definition_t* ebpf_map_definition, _Outptr_ ebpf_map_t** ebpf_map)
-{
-    ebpf_map_t* local_map = NULL;
-    size_t type = ebpf_map_definition->type;
-
-    if (ebpf_map_definition->type > BPF_MAP_TYPE_ARRAY)
```

Comment:
FYI to reviews: these functions were moved down later in this file after ebpf_map_function_tables[] is initialized so that this check can use _countof instead of having to be modified every time a new map type is added.